### PR TITLE
cmds/core/dd: Refactor code, add tests and push code into pkg/progress

### PR DESCRIPTION
**Pull request by walterchris**
This commits refactors a majority of the code within cmds/core/dd in
order to make it testable. Additionally it moves part of the code into
it's own package (pkg/progress) and adds tests for this package, and the
cmd itself.

Signed-off-by: Christian Walter <christian.walter@9elements.com>

### CODE REVIEW

**Review by rminnich — DISMISSED** [review]

**Comment by rminnich** [comment]
any idea why I'm not able to push a button and get a rebase? This is weird.

**Comment by walterchris** [comment]
Not sure.. I rebased and forced-pushed it.

File: cmds/core/dd/dd.go [hunk 1]
```diff
@@ -511,5 +451,5 @@ func main() {
 		log.Fatal(err)
 	}
 
-	progress.end()
+	progress.End()
```
Comment (rminnich) [inline cmds/core/dd/dd.go:454]: would it make sense to call it Close?

**Review by rminnich — APPROVED** [review]

File: cmds/core/dd/dd.go (hunk 1 above)
Comment (walterchris) [reply cmds/core/dd/dd.go:454]: Hm - this marks the end of the process. I find it "End" fits more than "Close".

**Pull request merged**